File: packages/expo-scroll-forwarder/ios/ExpoScrollForwarderModule.swift
import ExpoModulesCore

public class ExpoScrollForwarderModule: Module {
  // Each module class must implement the definition function. The definition consists of components
  // that describes the module's functionality and behavior.
  // See https://docs.expo.dev/modules/module-api for more details about available components.
  public func definition() -> ModuleDefinition {
    // Sets the name of the module that JavaScript code will use to refer to the module. Takes a string as an argument.
    // Can be inferred from module's class name, but it's recommended to set it explicitly for clarity.
    // The module will be accessible from `requireNativeModule('ExpoScrollForwarder')` in JavaScript.
    Name("ExpoScrollForwarder")

    // Enables the module to be used as a native view. Definition components that are accepted as part of the
    // view definition: Prop, Events.
    View(ExpoScrollForwarderView.self) {
      // Defines a setter for the `url` prop.
      Prop("url") { (view: ExpoScrollForwarderView, url: URL) in
        if view.webView.url != url {
          view.webView.load(URLRequest(url: url))
        }
      }

      Events("onLoad")
    }
  }
}

File: packages/expo-scroll-forwarder/android/src/main/java/expo/modules/scrollforwarder/ExpoScrollForwarderModule.kt
package expo.modules.scrollforwarder

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition
import java.net.URL

class ExpoScrollForwarderModule : Module() {
    override fun definition() = ModuleDefinition {
        Name("ExpoScrollForwarder")

        View(ExpoScrollForwarderView::class) {
            Prop("scrollViewTag") { view: ExpoScrollForwarderView, scrollViewTag: Int? ->
                view.scrollViewTag = scrollViewTag
            }
        }
    }
}
